datasets/pubmed/pubmed.py

```diff
@@ -0,0 +1,399 @@
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        dl_dir = dl_manager.download_and_extract(_URLs)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filenames": dl_dir},
+            ),
+        ]
+
+    def update_citation(self, article):
+        """
+        ArticleId and ArticleIdList are already used field name so we rewrite and
+        flatten those as {Citation, CitationId}.
+        """
+        citations = []
+        try:
+            list_ = article["PubmedData"]["ReferenceList"]
+        except Exception:
+            return
+
+        for ref in list_:
+            if "Reference" not in ref:
+                continue
+            for re in ref["Reference"]:
+                if "Citation" not in re:
+                    continue
+                citation = re["Citation"]
+                if "ArticleIdList" not in re:
+                    continue
+                for r in re["ArticleIdList"]:
+                    if "ArticleId" not in r:
+                        continue
+                    for rr in r["ArticleId"]:
+                        try:
+                            citation = {"Citation": citation, "CitationId": int(rr)}
+                        except Exception:
+                            continue
+                        citations.append(citation)
+        article["PubmedData"]["ReferenceList"] = citations
+
+    def _generate_examples(self, filenames):
+        """ Yields examples. """
+        id_ = 0
+        for filename in filenames:
+            print(filename)
+            try:
+                tree = etree.parse(filename)
+                root = tree.getroot()
+                xmldict = self.xml_to_dictionnary(root)
+            except etree.ParseError:
+                logger.error(f"Ignoring file {filename}, it is malformed")
+                continue
+
+            for article in xmldict["PubmedArticleSet"]["PubmedArticle"]:
+                self.update_citation(article)
+                new_article = default_article()
+
+                try:
+                    deepupdate(new_article, article)
+                except Exception:
+                    logger.error(f"Ignoring article {article}, it is malformed")
```

Comment:
Maybe use warning as well here ?

Showing an error should make the user take an action to fix it but it looks like there's not must we can do.

On the other side warnings are messages that the user may want to know about because it can affect the behavior of the program, while not requiring any action to be taken if the user is fine with it.

Reply:
Good point !